modules/core/src/test/java/org/apache/ignite/internal/processors/diagnostic/DiagnosticProcessorTest.java

```diff
@@ -167,24 +192,237 @@ public void testWalDirs() throws Exception {
     }
 
     /**
-     * Check that when an CorruptedTreeException is thrown, a "corruptedPages_TIMESTAMP.txt"
-     * will be created and a warning will be in the log.
+     * Simple POJO to generate binary metadata in {@link #doTestOutputDiagnosticCorruptedPagesInfo(Consumer, BiConsumer)}
+     */
+    public static class TestValue {
+        public final String val;
+
+        public TestValue(String val) {
+            this.val = val;
+        }
+    }
+
+    /**
+     * Check that when an CorruptedTreeException is thrown, a "corruptedPages_TIMESTAMP.txt" will be created, data will
+     * be dumped and a warning will be in the log.
      *
      * @throws Exception If failed.
      */
     @Test
     public void testOutputDiagnosticCorruptedPagesInfo() throws Exception {
+        doTestOutputDiagnosticCorruptedPagesInfo(cfg -> {}, (n, anyPageId) -> {
+            validateDiagnosticPathDir(n);
+
+            File baseDumpDir = getBaseDumpDir(n);
+            assertEquals(7, baseDumpDir.list().length);
+
+            // 6 folders to validate.
+            validateCacheDir(anyPageId, baseDumpDir);
+            validateWalDir(baseDumpDir);
+            validateBinaryMetaDir(baseDumpDir);
+            validateCpDir(baseDumpDir);
+            validateMetaStorageDir(baseDumpDir);
+            validateUtilityCacheDir(baseDumpDir);
+            validateLogsCacheDir(baseDumpDir);
+        });
+    }
+
+    /**
+     * Check that when an CorruptedTreeException is thrown, a "corruptedPages_TIMESTAMP.txt" will be created, data will
```

Comment:
This javadoc is the same as the javadoc of the previous test, but the test is different. Should this be reflected in the javadoc as well?

Reply:
Sure